ipemu/csrc/vbridge_impl.cc

```diff
@@ -377,34 +380,44 @@ std::optional<SpikeEvent> VBridgeImpl::spike_step() {
         .with("pc", fmt::format("{:08X}", se.pc))
         .with("rs1", fmt::format("{:08X}", se.rs1_bits))
         .with("rs2", fmt::format("{:08X}", se.rs2_bits))
+        .with("spike_cycles", spike_cycles)
         .info("spike run vector insn");
     se.pre_log_arch_changes();
-    pc = fetch.func(&proc, fetch.insn, state->pc);
+    new_pc = fetch.func(&proc, fetch.insn, state->pc);
     se.log_arch_changes();
   } else {
     Log("SpikeStep")
-        .with("pc", fmt::format("{:08X}", pc))
+        .with("pc", fmt::format("{:08X}", state->pc))
         .with("bits", fmt::format("{:08X}", fetch.insn.bits()))
         .with("disasm", proc.get_disassembler()->disassemble(fetch.insn))
+        .with("spike_cycles", spike_cycles)
         .info("spike run scalar insn");
-    pc = fetch.func(&proc, fetch.insn, state->pc);
+    new_pc = fetch.func(&proc, fetch.insn, state->pc);
   }
 
   // Bypass CSR insns commitlog stuff.
-  if ((pc & 1) == 0) {
-    state->pc = pc;
+  if ((new_pc & 1) == 0) {
+    state->pc = new_pc;
   } else {
-    switch (pc) {
+    Log("SpikeStep")
+      .with("new_pc", fmt::format("{:08X}", new_pc))
+      .with("priv", state->prv)
+      .with("ext", proc.extension_enabled_const('U'))
+      .info("CSR serialize");
+    switch (new_pc) {
     case PC_SERIALIZE_BEFORE:
       state->serialized = true;
       break;
     case PC_SERIALIZE_AFTER:
       break;
     default:
-      FATAL(fmt::format("SpikeStep: invalid pc: {:08X}", pc));
+      FATAL(fmt::format("SpikeStep: invalid new_pc: {:08X}", new_pc));
     }
   }
 
+  // spike does not bump mcycle by itself, so do it manually
```

Comment:
Maybe we should bump `spike_cycles` first and then record it to `mcycle`, or initialize `spike_cycles` with 1.

Reply:
You mean `mcycle` should start with 1? I did not expect that. May I ask for reasons?

Reply:
sorry, I misunderstand the difference between `spike_cycle` and `mcycle`, you are right.
review passed.